### ISSUE TITLE
Unable to build python with lzma support - Mojave with Macports

### ISSUE THREAD

**Issue by decuser** [pyenv/pyenv#1887]
I am trying to build python 3.9.1 on MacOS 10.14.6. This has worked in the past, but now that I'm trying to build with xz, etc., from macports, it is failing. Here's what I have done that worked in the past:

`PYTHON_CONFIGURE_OPTS="--with-tcltk-includes='-I/opt/local/includes' --with-tcltk-libs='-L/opt/local/lib/tcl8.6:/opt/local/lib/tk8.6 -ltcl8.6 -ltk8.6'" pyenv install 3.9.4`

This no longer works. It errors out with: `ModuleNotFoundError: No module named 'pyexpat'`
and complains about renaming pyexpat. I've read up on it and tried all of the suggestions - reinstall xcode CLT, add SDKROOT to build encantation, etc.

What works it to unset CFLAGS and LDFLAGS. That'd be swell, but now I want lzma stuff, which requires xz and such. So, after I `sudo port install xz zlibs etc`, I need a way to include the include files and the lib files from /opt/local... Normally, I would expect the following to work:

> export CFLAGS="$CFLAGS -I/opt/local/include"
  export LDFLAGS="$LDFLAGS -L/opt/local/lib"
  PYTHON_CONFIGURE_OPTS="--with-tcltk-includes='-I/opt/local/includes' --with-tcltk-libs='-L/opt/local/lib/tcl8.6:/opt/local/lib/tk8.6 -ltcl8.6 -ltk8.6'" pyenv install 3.9.4

But, then I'm back to the pyexpat issue. 

The reason I'm sticking to Mojave and using Macports in the first place : As some of you will begin to realize, brew will drop support for Mojave soon, whereas macports will continue to support ancient versions of Mac until people grow weary of doing so (those of us continuing to need 32 bit apps can't/won't upgrade to newer versions of MacOS).

**Comment by decuser** [comment]
Btw, I worked around the issue using a script that I found and modified for my own use during my investigations (I've lost track of where I found it, sorry). Here is what worked. I haven't had time to figure out what part of the incantation is the effective bit, but I present it here for posterity and any poor soul who can benefit from it:

```
vi pyenv-install.sh
#!/bin/bash
set +x

PY_VERSION=3.9.4

export MAKE_OPTS="-j8"

export PYTHON_CONFIGURE_OPTS="$PYTHON_CONFIGURE_OPTS --enable-framework"
export PYTHON_CONFIGURE_OPTS="$PYTHON_CONFIGURE_OPTS --with-computed-gotos"
export PYTHON_CONFIGURE_OPTS="$PYTHON_CONFIGURE_OPTS --with-system-expat"
export PYTHON_CONFIGURE_OPTS="$PYTHON_CONFIGURE_OPTS --with-system-ffi"
export PYTHON_CONFIGURE_OPTS="$PYTHON_CONFIGURE_OPTS --with-tcltk-includes='-I/opt/local/includes' --with-tcltk-libs='-L/opt/local/lib/tcl8.6:/opt/local/lib/tk8.6 -ltcl8.6 -ltk8.6'"
#export PYTHON_CONFIGURE_OPTS="$PYTHON_CONFIGURE_OPTS --with-ensurepip=no"
#export PYTHON_CONFIGURE_OPTS="$PYTHON_CONFIGURE_OPTS --enable-ipv6"

# optimizations
export CFLAGS="$CFLAGS -march=native"
export PYTHON_CONFIGURE_OPTS="$PYTHON_CONFIGURE_OPTS --enable-optimizations"
#export PYTHON_CONFIGURE_OPTS="$PYTHON_CONFIGURE_OPTS --with-lto"

export CFLAGS="$CFLAGS -I$HOME/.pyenv/versions/$PY_VERSION/openssl/include"
export CFLAGS="$CFLAGS -I$HOME/.pyenv/versions/$PY_VERSION/readline/include"
export CFLAGS="$CFLAGS -I/opt/local/include"

export LDFLAGS="$LDFLAGS -L$HOME/.pyenv/versions/$PY_VERSION/openssl/lib"
export LDFLAGS="$LDFLAGS -L$HOME/.pyenv/versions/$PY_VERSION/readline/lib"
export LDFLAGS="$LDFLAGS -L/opt/local/lib"

export PKG_CONFIG=`which pkg-config`

pyenv install -v --force $PY_VERSION
# end of file

sudo bash pyenv-install.sh
````

**Comment by native-api** [comment]
Looks like a duplicate of https://github.com/pyenv/pyenv/issues/1066

**Comment by decuser** [comment]
I disagree. Unless you are suggesting that pyenv depends on homebrew, rather than xz.

**Comment by decuser** [comment]
> Looks like a duplicate of #1066

#1068  refers to what appears to be a dependency on homebrew, not xz. This issue is not about homebrew. Unless you are suggesting that pyenv depends on homebrew explicitly?

**Comment by native-api** [comment]
Pyenv supports locating dependencies in Homebrew and also Zlib in XCode SDK. If you've installed them some other way and they require additional switches for Python's build to locate them, it's your job to supply those.

Check https://github.com/pyenv/pyenv/wiki/Common-build-problems

**Comment by native-api** [comment]
The root cause is likely earlier in the build: `pyexpat` is absent because it could not be built.

You didn't supply the debug output as required by the issue template so I cannot say anything more anyway.

**Comment by decuser** [comment]
Thanks for the assist.

**Comment by decuser** [comment]
Ah. If you read the update, you would see that I found a workaround. Not that I understand what it did, exactly, just that it worked, repeatedly :). So, I can live with it as is. I use pyenv on linux, freebsd, and windows. It’s exceptionally convenient. So when it wouldn’t build, I was a little panicked. With the workaround, it’s a little extra work, but doable. When I have time, I’ll go back and try to understand it better. 

Thanks for the template tip - I should have seen that! I captured the output, but figured it was overkill. Next time, I’ll read the post guidelines better.

Will
